src/test/java/org/apache/sysds/test/functions/builtin/part1/BuiltinHMMPredictTest.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.sysds.test.functions.builtin.part1;
+
+import org.apache.sysds.api.DMLScript;
+import org.apache.sysds.common.Types;
+import org.apache.sysds.common.Types.ExecType;
+import org.apache.sysds.hops.OptimizerUtils;
+import org.apache.sysds.test.AutomatedTestBase;
+import org.apache.sysds.test.TestConfiguration;
+import org.apache.sysds.test.TestUtils;
+import org.junit.Test;
+
+@net.jcip.annotations.NotThreadSafe
+public class BuiltinHMMPredictTest extends AutomatedTestBase
+{
+	private final static String TEST_NAME = "hmmPredict";
+	private final static String TEST_DIR = "functions/builtin/";
+	private static final String TEST_CLASS_DIR = TEST_DIR + BuiltinHMMPredictTest.class.getSimpleName() + "/";
+	private final static int rows = 1320;
+	private final static int cols = 32;
+
+	@Override
+	public void setUp() {
+		TestUtils.clearAssertionInformation();
+		addTestConfiguration(TEST_NAME,new TestConfiguration(TEST_CLASS_DIR, TEST_NAME,new String[]{"C"}));
+	}
+    	
+    @Test    
+	public void testHMMPredictCP() {
+		runHMMPredictTest(ExecType.CP);
+	}
+
+	@Test    
+	public void testHMMPredictSingleCP() {
+		runHMMPredictTest(ExecType.SPARK);
+	}
+	
+	private void runHMMPredictTest(ExecType instType)
+	{
+		Types.ExecMode platformOld = setExecMode(instType);
+
+		boolean oldFlag = OptimizerUtils.ALLOW_ALGEBRAIC_SIMPLIFICATION;
+		boolean sparkConfigOld = DMLScript.USE_LOCAL_SPARK_CONFIG;
+
+		try
+		{
+			loadTestConfiguration(getTestConfiguration(TEST_NAME));
+
+			String HOME = SCRIPT_DIR + TEST_DIR;
+
+			fullDMLScriptName = HOME + TEST_NAME + ".dml";
+			programArgs = new String[]{
+					"-nvargs", "X=" + input("X"), "k=" + inpu("k"), "Y=" + output("Y")};
+
+
+			"""			
+			//generate actual datasets
+			double[][] X = getRandomMatrix(rows, cols);
+			Integer k = 
+			writeInputMatrixWithMTD("X", X, true);
+			"""
+
+			runTest(true, false, null, -1);
```

Comment:
this only runs your code. it does not verify something happened.

you can alternatively call it like 


String stdout = runTest(null).toString()

then you get the output from the experiment if you also call before setOutputBuffering(true);